packages/ui/src/components/Menu/index.tsx

```diff
@@ -183,15 +193,26 @@ const FwdMenu = forwardRef(
     })
 
     // if you need dialog inside your component, use function, otherwise component is fine
-    const target = isValidElement(disclosure) ? disclosure : disclosure(popover)
+    const target = isValidElement<ButtonHTMLAttributes<HTMLButtonElement>>(
+      disclosure,
+    )
+      ? disclosure
+      : disclosure(popover)
     const innerRef = useRef(target as unknown as HTMLButtonElement)
     useImperativeHandle(ref, () => innerRef.current)
 
     return (
       <>
         {disclosure && (
           // @ts-expect-error reakit types are invalid, no need to pass as something, default is div
-          <PopoverDisclosure {...popover} ref={innerRef}>
+          <PopoverDisclosure
+            {...popover}
+            onClick={(e: MouseEvent<HTMLButtonElement>) => {
+              // This allow to use onClick for speficique case like traking.
+              target?.props?.onClick?.(e)
```

Comment:
```suggestion
            onClick={(event: MouseEvent<HTMLButtonElement>) => {
              // This allow to use onClick for specific case like tracking.
              target?.props?.onClick?.(event)
```

we avoid using diminutive letters and use the full name of the variable

Reply:
@philibea made the mistake also, can you change it here too: `packages/ui/src/components/Modal/index.tsx`?